composer/utils/checkpoint.py

```diff
@@ -255,17 +279,74 @@ def _download_checkpoint(
     return composer_states_filepath, extracted_checkpoint_folder, extracted_rank_n
 
 
+def glob_filter(exclude_globs: List[str]) -> Callable[[Dict], None]:
+    """Provides a function which deletes all subparts of a dictionary based on a list of paths."""
+
+    def filter_func(state_dict: Dict) -> None:
+
+        def flatten_state_dict(state_dict: Any, paths: List[str], existing_path: str):
+            """Recursively convert a dictionary into a set of paths."""
+            # Store path when we reach end, which is either non-Dict or empty Dict
+            if not isinstance(state_dict, Dict) or len(state_dict.keys()) == 0:
+                # Remove leading /
+                paths.append(existing_path.lstrip('/'))
+            # Descend down all keys
+            else:
+                for key in state_dict.keys():
+                    flatten_state_dict(state_dict[key], paths, f"{existing_path}/{key}")
+
+        # Flatten dictionary into paths
+        paths = []
+        flatten_state_dict(state_dict, paths, '/')
+        print(paths)
+
+        filtered_paths = []
+        for exclude_glob in exclude_globs:
+            filtered_paths_from_glob = fnmatch.filter(paths, exclude_glob)
+            if len(filtered_paths_from_glob) == 0:
+                log.warning(
+                    f"No parts from loaded checkpoint state_dict were ignored by load_ignore_key {exclude_glob}")
+            filtered_paths.extend(filtered_paths_from_glob)
+        filtered_paths = list(set(filtered_paths))
+        filtered_paths_str = " ".join(filtered_paths)
+        if len(filtered_paths_str) > 0:
+            log.info(f"Ignoring the following paths from the loaded checkpoint state_dict: {filtered_paths_str}")
+
+        def remove_path(state_dict: Dict, exclude_glob: List[str]):
```

Comment:
Same comments as above re: moving this into a private helper and supporting lists.

Reply:
Done.